File: src/main/java/tfg/proyecto/TFG/dtos/DTOeventoSubida.java
package tfg.proyecto.TFG.dtos;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Singular;
import tfg.proyecto.TFG.modelo.Categoria;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DTOeventoSubida {
		
		private Long id;
	 	private String nombre;
	    private String localizacion;
	    private LocalDateTime inicioEvento;
	    private LocalDateTime finEvento;
	    private String descripcion;
	    private double precio;
	    private int aforo;
	    private Categoria categoria;


	    private String imagen; 
	    
	    @Singular
	    private List<String> imagenesCarrusels;


	    private List<DTOInvitadoSubida> invitados;
	    private Long vendedorId;

          

}

File: frontend/appMovilShowPass/app/src/main/java/com/example/appmovilshowpass/data/remote/dto/DTOeventoSubida.kt
package com.example.appmovilshowpass.data.remote.dto

import com.example.appmovilshowpass.model.Categoria
import com.example.appmovilshowpass.model.Evento
import com.example.appmovilshowpass.model.Invitado
import com.example.appmovilshowpass.utils.construirUrlImagen

data class DTOeventoSubida(
    val id: Long? = null,
    val nombre: String,
    val localizacion: String,
    val inicioEvento: String,
    val finEvento: String,
    val descripcion: String,
    val precio: Double,
    val categoria: Categoria,
    val aforoMax: Int,
    val vendedorId: Long,
    val imagen: String?,
    val imagenesCarruselUrls: List<String>?,
    val invitados: List<DTOInvitadoSubida>?

    )
